### PR TITLE
Ad Hoc: Performance sanity CI improvements

### PR DESCRIPTION
**Pull request by shailesh-vaidya**
# Problem Statement
- Improve performance-sanity CI with the following features 

1.   Added support for HW. We are still facing issues with HW node deployment so not enabling nightly yet. 
2.   Fetch S3 credentials from deployed cluster instead of solution.yaml. 

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM
- [x] Jenkins pipelines used for testing - https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/Performance/job/nightly-performance-ci/108/

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: jenkins/performance-ci/nightly-performance-ci.groovy [hunk 1]
```diff
@@ -61,26 +69,65 @@ pipeline {
...
-                        string(name: 'EXTERNAL_EXPOSURE_SERVICE', value: "NodePort"),
-                        string(name: 'SNS_CONFIG', value: "${SNS_CONFIG}"),
-                        string(name: 'DIX_CONFIG', value: "${DIX_CONFIG}")
-                    ]
-                    env.cortxcluster_build_url = cortxCluster.absoluteUrl
-                    env.cortxCluster_status = cortxCluster.currentResult
-                }
-            }
+                    if (params.infrastructure == "HW") {
+                        sh label: 'Copy RPMS', script:'''
```
Comment (gauravchaudhari02) [inline jenkins/performance-ci/nightly-performance-ci.groovy:75]: please change label name suitable for action

File: jenkins/performance-ci/nightly-performance-ci.groovy [hunk 2]
```diff
@@ -61,26 +69,65 @@ pipeline {
...
+                        echo "Deploying CORTX on Hardware"
+                        '''
+                        checkout([$class: 'GitSCM', branches: [[name: '$CORTX_RE_BRANCH']], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'AuthorInChangelog']], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'cortx-admin-github', url: '$CORTX_RE_REPO']]])
+
+                        sh label: 'Copy RPMS', script:'''
+                        pushd solutions/kubernetes/
+                            curl -l https://raw.githubusercontent.com/shailesh-vaidya/cortx-re/hw-performance-ci/jenkins/performance-ci/nightly-performance-ci.yml -o solution.yaml
```
Comment (gauravchaudhari02) [inline jenkins/performance-ci/nightly-performance-ci.groovy:82]: please change repository to main repository if test is completed

File: jenkins/performance-ci/nightly-performance-ci.groovy [hunk 3]
```diff
@@ -61,26 +69,65 @@ pipeline {
...
-                    env.cortxCluster_status = cortxCluster.currentResult
-                }
-            }
+                    if (params.infrastructure == "HW") {
+                        sh label: 'Copy RPMS', script:'''
+                        echo "Deploying CORTX on Hardware"
+                        '''
+                        checkout([$class: 'GitSCM', branches: [[name: '$CORTX_RE_BRANCH']], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'AuthorInChangelog']], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'cortx-admin-github', url: '$CORTX_RE_REPO']]])
+
+                        sh label: 'Copy RPMS', script:'''
```
Comment (gauravchaudhari02) [inline jenkins/performance-ci/nightly-performance-ci.groovy:80]: please add appropriate label name

File: jenkins/performance-ci/nightly-performance-ci.groovy [hunk 4]
```diff
@@ -61,26 +69,65 @@ pipeline {
...
+                            #Destroy CORTX Cluster
+                            ls -ltr
+                            ./cortx-deploy.sh --destroy-cluster
+
+                            #Deploy CORTX Cluster
+                            ./cortx-deploy.sh --cortx-cluster
+
+                        '''
+                    } else {
+                        sh label: 'Copy RPMS', script:'''
```
Comment (gauravchaudhari02) [inline jenkins/performance-ci/nightly-performance-ci.groovy:106]: please add appropriate label name

File: jenkins/performance-ci/performance-sanity.groovy [hunk 5]
```diff
@@ -44,6 +44,12 @@ pipeline {
                         export GITHUB_TOKEN=${GITHUB_CRED}
                         export CORTX_TOOLS_REPO=${CORTX_TOOLS_REPO}
                         export CORTX_TOOLS_BRANCH=${CORTX_TOOLS_BRANCH}
+                        export DB_SERVER=10.237.65.111
```
Comment (gauravchaudhari02) [inline jenkins/performance-ci/performance-sanity.groovy:47]: is it safe to expose IP?

File: jenkins/performance-ci/nightly-performance-ci.groovy (hunk 1 above)
Comment (shailesh-vaidya) [reply jenkins/performance-ci/nightly-performance-ci.groovy]: Fixed

File: jenkins/performance-ci/nightly-performance-ci.groovy (hunk 3 above)
Comment (shailesh-vaidya) [reply jenkins/performance-ci/nightly-performance-ci.groovy]: Fixed

File: jenkins/performance-ci/nightly-performance-ci.groovy (hunk 4 above)
Comment (shailesh-vaidya) [reply jenkins/performance-ci/nightly-performance-ci.groovy]: Fixed

File: jenkins/performance-ci/nightly-performance-ci.groovy (hunk 2 above)
Comment (shailesh-vaidya) [reply jenkins/performance-ci/nightly-performance-ci.groovy]: fixed

File: jenkins/performance-ci/performance-sanity.groovy (hunk 5 above)
Comment (shailesh-vaidya) [reply jenkins/performance-ci/performance-sanity.groovy]: Removed hardcoded IP details

File: jenkins/performance-ci/nightly-performance-ci.yml [hunk 6]
```diff
@@ -0,0 +1,271 @@
+solution:
+  namespace: default
```
Comment (gauravchaudhari02) [inline jenkins/performance-ci/nightly-performance-ci.yml:2]: Should we change it to `cortx`?

File: jenkins/performance-ci/performance-check.groovy [hunk 7]
```diff
@@ -86,19 +92,87 @@ pipeline {
...
+        stage ('Deploy CORTX Cluster on HW') {
+             when {
+                expression { params.infrastructure == 'HW' }
+            }
+            steps {
+                script { build_stage = env.STAGE_NAME }
+
+                checkout([$class: 'GitSCM', branches: [[name: '$CORTX_RE_BRANCH']], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'AuthorInChangelog']], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'cortx-admin-github', url: '$CORTX_RE_REPO']]])
+
+                sh label: 'Copy RPMS', script:'''
```
Comment (gauravchaudhari02) [inline jenkins/performance-ci/performance-check.groovy:143]: Please add appropriate stage name

File: jenkins/performance-ci/nightly-performance-ci.yml (hunk 6 above)
Comment (shailesh-vaidya) [reply jenkins/performance-ci/nightly-performance-ci.yml]: Removed this file

File: jenkins/performance-ci/performance-check.groovy (hunk 7 above)
Comment (shailesh-vaidya) [reply jenkins/performance-ci/performance-check.groovy]: Fixed

**Review by gauravchaudhari02 — APPROVED** [review]

**Pull request merged**